plugins/hls-refactor-plugin/src/Development/IDE/Plugin/CodeAction.hs

```diff
@@ -438,19 +438,25 @@ suggestRemoveRedundantImport ParsedModule{pm_parsed_source = L _  HsModule{hsmod
         = [("Remove import", [TextEdit (extendToWholeLineIfPossible contents _range) ""])]
     | otherwise = []
 
+diagInRange :: Diagnostic -> Range -> Bool
+diagInRange Diagnostic {_range = dr} r = dr `subRange` extendedRange
+  where
+    -- Ensures the range captures full lines. Makes it easier to trigger the correct
+    -- "remove redundant" code actions from anywhere on the offending line.
+    extendedRange = extendToFullLines r
```

Comment:
I am a bit unsure about this, so this makes it easier for code actions to overlap, right?

So if we have two refactor code actions in the same line, they'd overlap, potentially making it unclear which code action would be truly triggered for an diagnostic?

Reply:
> so this makes it easier for code actions to overlap, right?

It does! But (at least in my opinion) in a good way :)

For example, here are two lines with unused imports:

![Screenshot from 2024-02-14 10-58-18](https://github.com/haskell/haskell-language-server/assets/132365/065a0739-adc2-4ee2-a339-a020f7f0ffb0)

*With* extending to full line, I can move my cursor to *anywhere* on the offending `import` line -- say, the beginning of the line -- and show code actions, and I'll see all code actions that are relevant to that `import`, including ones about redundancies:

![Screenshot from 2024-02-14 11-01-48](https://github.com/haskell/haskell-language-server/assets/132365/b9f9c9dd-8492-4cee-ba03-b0c7b8a49820)

*Without* extending to full line, I need to put my cursor *directly on* `showInfoBrief` to see the relevant code actions. Which also makes things a bit confusing, IMO... especially since some import-related code actions ("Make imports explicit", e.g.) aren't tied to diagnostics at all, it can seem a little random for a user to click on a different part of an import and see a different set of code actions.

> So if we have two refactor code actions in the same line, they'd overlap, potentially making it unclear which code action would be truly triggered for an diagnostic?

There's always the possibility of overlap with code actions regardless, right? Multiple actions can apply to a single piece of code. But it's not a problem -- the code actions still work exactly as expected, and it's still up to the user to select what they'd like to actually *change*. For example, on the unused `import Debug.Trace` line, I get other actions in addition to the redundant import ones:

![Screenshot from 2024-02-14 11-05-43](https://github.com/haskell/haskell-language-server/assets/132365/4adf54c8-7f2e-4e82-b3ad-d4123d088cad)

And they all work great, as expected.

To me, extending to the full line in *this particular case* -- with redundant imports -- seemed like a usability improvement. And in fact, as a user I'd be a little surprised if it didn't work that way.

But of course I'm open to changing it if folks feel otherwise!